### ISSUE TITLE
Fix lyrics scraping on genius.com

### ISSUE THREAD

**Issue by wummel** [beetbox/beets#3727]
### Problem

The lyrics plugin cannot get the lyric text on some pages of genius.com.
Example:
https://genius.com/Ed-sheeran-nothing-on-you-lyrics
The plugin expects this:
```html
<div class="lyrics">
```
When running 
```sh
$ beet -vv lyrics
```
the above page will result in:

```
Traceback (most recent call last):
[...]
File "/usr/lib/python3.7/site-packages/beetsplug/lyrics.py", line 375, in lyrics_from_song_api_path
    lyrics = html.find("div", class_="lyrics").get_text()
AttributeError: 'NoneType' object has no attribute 'get_text'
```
Reason is that genious.com seems to have several pages where instead of a single div the
text is stored in a list of divs where the class name starts with Lyrics_Container:
```html
<div class="Lyrics__Container-sc-1ynbvzw-2 jgQsqn">...</div>
<div class="Lyrics__Container-sc-1ynbvzw-2 jgQsqn">...</div>
<div class="Lyrics__Container-sc-1ynbvzw-2 jgQsqn">...</div>
```

### Setup

* OS: linux
* Python version: 3.7.3
* beets version: 1.4.9
* Turning off plugins made problem go away (yes/no): yes (lyrics plugin)

To reproduce, enable the lyrics plugin, set the genius as only importer and try to
scrape a fairly large music collection.

```yaml
lyrics:
    bing_lang_from: []
    auto: yes
    sources: genius
    bing_client_secret: REDACTED
    bing_lang_to:
    google_API_key: REDACTED
    google_engine_ID: REDACTED
    genius_api_key: REDACTED
    fallback:
    force: no
    local: no
directory: REDACTED/music
ignore_hidden: yes
asciify_paths: yes

import:
    move: no
    write: yes
    incremental: no
    resume: no

plugins: lyrics

```

Here is a patch that worked for me:

```diff
--- lyrics.py.orig      2020-08-12 20:10:01.000000000 +0200
+++ lyrics.py   2020-08-12 20:10:01.000000000 +0200
@@ -370,11 +370,21 @@
         # Remove script tags that they put in the middle of the lyrics.
         [h.extract() for h in html('script')]
 
-        # At least Genius is nice and has a tag called 'lyrics'!
-        # Updated css where the lyrics are based in HTML.
-        lyrics = html.find("div", class_="lyrics").get_text()
-
-        return lyrics
+        # Genius has the lyrics either in multiple divs with class attributes
+        # beginning with "Lyrics__Container", or in a single div with class
+        # attribute "lyrics"
+        lyric_tag = html.find("div", class_="lyrics")
+        if lyric_tag is None:
+            class_matcher = re.compile("^Lyrics__Container")
+            lyric_tags = html.find_all("div", class_=class_matcher)
+            if not lyric_tags:
+                self._log.debug(u'Genius page {0} has no lyric tags', page_url)
+                return None
+            lyrics = u'\n\n'.join(tag.get_text() for tag in lyric_tags)
+        else:
+            lyrics = lyric_tag.get_text()
+        # remove leading and trailing whitespace
+        return lyrics.strip()
 
     def fetch(self, artist, title):
         search_url = self.base_url + "/search"
```

**Comment by sampsyo** [comment]
Seems great! Would you mind transforming this patch into a pull request?

**Comment by 7h3qu1rkyb1t** [comment]
i am getting same error while importing from same file
`Correcting track tags from:
    Scavenger Hunt - Lost
To:
    Scavenger Hunt - Lost (Lenno remix)
URL:
    https://musicbrainz.org/recording/beccc97e-5a48-4108-ae3f-106fb94138d3
(Similarity: 94.6%) (title)
[A]pply, More candidates, Skip, Use as-is, Enter search, enter Id, aBort? a
Traceback (most recent call last):
  File "/usr/bin/beet", line 33, in <module>
    sys.exit(load_entry_point('beets==1.4.9', 'console_scripts', 'beet')())
  File "/usr/lib/python3.8/site-packages/beets/ui/__init__.py", line 1266, in main
    _raw_main(args)
  File "/usr/lib/python3.8/site-packages/beets/ui/__init__.py", line 1253, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/usr/lib/python3.8/site-packages/beets/ui/commands.py", line 955, in import_func
    import_files(lib, paths, query)
  File "/usr/lib/python3.8/site-packages/beets/ui/commands.py", line 925, in import_files
    session.run()
  File "/usr/lib/python3.8/site-packages/beets/importer.py", line 329, in run
    pl.run_parallel(QUEUE_SIZE)
  File "/usr/lib/python3.8/site-packages/beets/util/pipeline.py", line 445, in run_parallel
    six.reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/home/gireesh/.local/lib/python3.8/site-packages/six.py", line 703, in reraise
    raise value
  File "/usr/lib/python3.8/site-packages/beets/util/pipeline.py", line 312, in run
    out = self.coro.send(msg)
  File "/usr/lib/python3.8/site-packages/beets/util/pipeline.py", line 194, in coro
    func(*(args + (task,)))
  File "/usr/lib/python3.8/site-packages/beets/importer.py", line 1511, in plugin_stage
    func(session, task)
  File "/usr/lib/python3.8/site-packages/beets/plugins.py", line 143, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 826, in imported
    self.fetch_item_lyrics(session.lib, item,
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 840, in fetch_item_lyrics
    lyrics = [self.get_lyrics(artist, title) for title in titles]
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 840, in <listcomp>
    lyrics = [self.get_lyrics(artist, title) for title in titles]
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 873, in get_lyrics
    lyrics = backend.fetch(artist, title)
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 403, in fetch
    return self.lyrics_from_song_api_path(song_api_path)
  File "/usr/lib/python3.8/site-packages/beetsplug/lyrics.py", line 375, in lyrics_from_song_api_path
    lyrics = html.find("div", class_="lyrics").get_text()
AttributeError: 'NoneType' object has no attribute 'get_text'`

**Comment by schezfaz** [comment]
Hey, could I pick this issue up, and leverage the patch @wummel provided in order to have a sure shot solution?

**Comment by sampsyo** [comment]
That would be awesome!

**Comment by piotr-szegda** [comment]
I think it still not solved.

**Comment by sampsyo** [comment]
I believe that's true. If anybody has the bandwidth to take the above patch and open a quick PR with it, we can get the process of fixing things started!

**Comment by jackwilsdon** [comment]
I can take a look at picking this up.

**Comment by jackwilsdon** [comment]
This seems to already be fixed (as of #3594) - available in beets 1.5.0.
